### DISCUSSION TITLE
How to adjust font weight?

### DISCUSSION THREAD

**Discussion by 26558964** [alshedivat/al-folio#2394] [Q&A] [answered]
It seems that the theme is using 300 Roboto by default (400 for bold), and changing the font doesn't change the weight. So this is a global setting. But this seems a bit too light to me. Does any one have any idea how to change that, for regular, bold, etc?

**Comment by george-gca** [comment, marked as answer]
If you check our [_sass/_base.scss](https://github.com/alshedivat/al-folio/blob/master/_sass/_base.scss) we do some override for the font properties depending on the use case.

As for changing the Roboto fonts used, simply go to https://fonts.google.com/, select `Roboto`, click on `Get font` on the upper right, then `Get embed code`. In the new screen, on the left, you can click `Change styles` to select which versions of the font will be used. After selecting what you want simply copy the `<link>` embed code and replace it in [_config.yml](https://github.com/alshedivat/al-folio/blob/0e54bf6444e890607cdc02e112df5f9ac2f8ef76/_config.yml#L461).

**Reply by 26558964** [reply]
Thank you so much! It worked.

**Reply by glenntfung** [reply]
@george-gca @26558964  I didn't have those lines originally when I used the template. If I just copy the `library version` section of the answer's [_config.ylm](https://github.com/alshedivat/al-folio/blob/0e54bf6444e890607cdc02e112df5f9ac2f8ef76/_config.yml#L461) and adjust the font using the Google font link, the display math in the posts would break and become code wrapped by `\[\]` though it was wrapped by `$$ $$` in the markdown file. I don't think this is my code's problem because it was originally fine and the preview on GitHub is also correct. 

Example of broken display math: [this page](https://glenn-fung.github.io/blog/2023/cj/)
Corresponding markdown file: [this one](https://github.com/glenn-fung/glenn-fung.github.io/blob/master/_posts/2023-12-12-cj.md)

Do you guys have any idea why? Many thanks in advance.

**Reply by glenntfung** [reply]
I think maybe copying is not correct, as other problems also appear, for example, the “more” button of the navigation bar on iPhone (the button consists of three horizontal lines) doesn’t work as well. Could you please check how I can change the Roboto in my version? Thanks so much!

**Reply by george-gca** [reply]
First, almost every time is not something as simple as copying just a few modifications in a single file. The library versions in the `_config.yml` was created to change a whole behavior inside other parts of the code (see #2312), and if these aren't replicated along, then this piece in config alone won't do much.

Second, there is not much that I can do to help in your case but to point directions. Changing a font in the site is not always that easy. This site was thought to use Roboto with the specified sizes and types, so changing a font would sometimes alter too much the layouts, being some pieces too large, others too small, so it is really a matter of changing and adapting a lot of different settings in the site to make it look nice. Also, I don't have any Mac devices in my setup, so I can't even replicate the issue.

As for the navigation bar button, it is defined in [_includes/header.liquid](https://github.com/alshedivat/al-folio/blob/0e54bf6444e890607cdc02e112df5f9ac2f8ef76/_includes/header.liquid#L29), if you want to change it for your specific case.

**Reply by glenntfung** [reply]
OK, thanks!
